src/main/kotlin/uk/gov/gdx/datashare/config/WebClientConfiguration.kt

```diff
@@ -37,4 +42,47 @@ class WebClientConfiguration(
       .defaultHeader("X-Auth-Username", "gdx-data-share-user")
       .build()
   }
+
+  @Bean
+  @ConditionalOnProperty(name = ["api.base.prisoner-event.enabled"], havingValue = "true")
+  fun prisonerSearchApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("prisoner-search")
+
+    val connectionProvider = ConnectionProvider.builder("prisonerSearch")
+      .maxConnections(100)
+      .maxIdleTime(Duration.ofSeconds(20))
+      .maxLifeTime(Duration.ofSeconds(60))
+      .pendingAcquireTimeout(Duration.ofSeconds(60))
+      .evictInBackground(Duration.ofSeconds(120))
+      .build()
+
+    val httpClient = HttpClient.create(connectionProvider).responseTimeout(Duration.ofMinutes(10))
```

Comment:
can we extract the http client creation to a seperate function as it is the same for both webclients

Reply:
yes!